fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java

```diff
@@ -2085,8 +2086,22 @@ List<TExecPlanFragmentParams> toThrift(Set<TUniqueId> inFlightInstanceIds,
 
             WorkGroup workgroup = null;
             if (connectContext != null) {
-                workgroup = GlobalStateMgr.getCurrentState().getWorkGroupMgr().chooseWorkGroup(
-                        connectContext, WorkGroupClassifier.QueryType.SELECT, dbIds);
+                SessionVariable sessionVariable = connectContext.getSessionVariable();
+
+                if (StringUtils.isNotEmpty(sessionVariable.getResourceGroup())) {
+                    // Specify the resource group through variable
+                    String rgName = sessionVariable.getResourceGroup();
+                    WorkGroup wg = GlobalStateMgr.getCurrentState().getWorkGroupMgr().chooseWorkGroupByName(rgName);
```

Comment:
I think you should valid the the resource group name when set resource group.

Reply:
Maybe both, a resource group could be dropped when executing.

Reply:
done